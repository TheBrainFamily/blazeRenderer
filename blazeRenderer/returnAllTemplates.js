import glob from 'glob'

export default function(path) {

  const value = [path].map(function (path) {
    if (path.match(/.*\*\*\/\*.html/)) {
      return glob.sync(path, {cwd: process.cwd()})
    }

    if (path.match(/.*\.html/)) {
      return path
    }
    return glob.sync(path + '/**/*.html', {cwd: process.cwd()})
  }).reduce(function (a, b) {
    return a.concat(b)
  }, [])

  return value;
}