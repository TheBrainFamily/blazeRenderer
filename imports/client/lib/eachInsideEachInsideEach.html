<template name="eachInsideEachInsideEach">
    <div>
        {{#each forEachOneFirst}}
        first {{test}}
        {{#each forEachOneSecond}}
        second {{test}}
        {{#each forEachOneThird}}
        third {{test}}
        {{#each forEachOneFourth}}
        fourth {{test}}
        {{/each}}
        {{/each}}
        {{/each}}
        {{/each}}
    </div>
</template>