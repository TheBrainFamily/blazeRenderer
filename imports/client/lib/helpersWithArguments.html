<template name="helpersWithArguments">
    {{say "hello"}}
</template>