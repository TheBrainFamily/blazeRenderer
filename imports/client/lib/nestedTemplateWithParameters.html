<template name="parentTemplate">
    <div>Parent: {{> nestedWithParams param=hello test="not hello"}}</div>
</template>

<template name="nestedWithParams">
    <div>Child: {{param}} {{test}} Param Again - {{almostReactivelyWithParentParam}}</div>
</template>