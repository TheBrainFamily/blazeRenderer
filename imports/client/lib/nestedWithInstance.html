<template name="parentWithInstance">
    <div>{{fromParentTemplateInstance}}
        {{> nestedWithInstance }}
    {{fromParentTemplateInstance}}</div>
</template>

<template name="nestedWithInstance">
    <div>{{fromNestedTemplateInstance}}</div>
</template>