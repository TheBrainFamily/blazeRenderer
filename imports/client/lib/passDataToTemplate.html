<template name="passDataToTemplate">
    {{#with dataSource}}
        <div>
            should display "some value" below
            {{> templateToPassDataTo key.value}}
        </div>
    {{/with}}
</template>

<template name="templateToPassDataTo">
    should displayed passed argument:
    {{displayHelper this}}
</template>