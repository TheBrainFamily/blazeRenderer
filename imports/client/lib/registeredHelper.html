<template name="forRegisteredHelper">
    {{#if areEqual 'test' 'test'}}
    <div>
        <p>Register helper working properly!!!</p>
    </div>
    {{/if}}
</template>
