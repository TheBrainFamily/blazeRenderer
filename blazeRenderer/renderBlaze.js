import html from './blazeInternals/html'
import compile from './blazeInternals/compiler'
import fs from 'fs';
import { html as beautifyHtml } from 'js-beautify'
import './blazeInternals/blazeWith'
import './blazeInternals/template'
import './mockTemplates'
import './mockReactiveVariable'
import returnAllTemplates from './returnAllTemplates'

var toHTML = function (data, template, templateName, matchedInsideTemplates) {

    if (matchedInsideTemplates) {
      matchedInsideTemplates.forEach(insideTemplate => {
        var compiledInside = compile(insideTemplate.templateInside ,{isBody: true})
        var fnInside = eval(compiledInside)
        window.hackedInTemplates = {}
        window.hackedInTemplates[insideTemplate.name] = fnInside
      })
    }

    var compiled = compile(template, {isBody: true});
    var fn = eval(compiled)
    return BlazeMine.toHTML(BlazeMine.With(data, fn, templateName));
};

export const renderBlazeWithData = function renderBlaze(templateFile, templateName, data) {
    const include = function includeReplacement(templateName, data) {
        data = data || {};
        data = Object.assign({}, data, {includeReplacement})

		return toHTML(data, $(`template[name='${templateName}']`).html().toString().replace(/&gt;/g, ">").replace(/&apos;/g, "'").replace(/&quot;/g, '"'), templateName);
    }

    const template = fs.readFileSync(templateFile)
    var cheerio = require('cheerio');
    var $ = cheerio.load(template.toString().replace(/{{> *([^\s}]*)([^}]*)}}/g, '{{{ includeReplacement \'$1\' $2 }}}'));

    return beautifyHtml(include(templateName, data))
}

export const parseTemplates = function (templateFiles) {
  const templatesToFilesMap = []
  templateFiles.forEach((templateFile) => {
    const template = fs.readFileSync(templateFile)
    //TODO add test cases for multiline {{> }}
    //TODO add test case for a case when {{> were}} <- no space
    const parsedText = template.toString().replace(/{{(>) ?(Template.contentBlock)/g, '{{ $2').replace(/{{> *([^\s}]*)([^}]*)}}/g, '{{{ includeReplacement \'$1\' $2 }}}').replace(/({{|{{.*( |=))(this)( |}})/g, '$1_myOwnThis$4').replace(/&gt;/g, '>').replace(/&apos;/g, '\'').replace(/&quot;/g, '"')

    const templateRegex = /<template name=("|')(.*)("|')>((.|\n)*?)<\/template>/gm

    let match = templateRegex.exec(parsedText)

    while (match !== null) {
      templatesToFilesMap.push({templateName: match[2].trim(), templateFile, cheerio: match[4].trim()})
      match = templateRegex.exec(parsedText)
    }

  })
    return templatesToFilesMap;
}

const renderBlazeWithTemplates = function (templateName, parsedTemplates) {
  if (parsedTemplates && parsedTemplates.length > 0) {

  } else {
    parsedTemplates = parseTemplates(returnAllTemplates('imports/').concat(returnAllTemplates('client/')))
  }
    const includeReplacement = function includeReplacement(templateName) {
    let passedArguments;
    if (Array.from(arguments)[1] ) {
      passedArguments = Array.from(arguments)[1]['hash']  ? Array.from(arguments)[1]['hash'] : {_myOwnData: Array.from(arguments)[1]}
    }
        Template[templateName].helpers = Object.assign({}, Template[templateName].getHelpers(), passedArguments, {isInRole: function() { return true }}, {$or: function(arg1, arg2) { return arg1 || arg2}}, {$gt: function(arg1, arg2) { return arg1 > arg2}}, {$eq: function(arg1, arg2) { return arg1 === arg2 }}, {$exists: function(arg1) { return !!arg1 }}, {pathFor: function(arg1, arg2) { return `${arg1}/${arg2}`}}, {_myOwnThis: function() {return this._myOwnData}}, {$:{Session:{get: function(arg) { return true }}}})

      //TODO add test for isInRole, and most probably make this configurable instead of hardcoded.
      // Used in https://github.com/alanning/meteor-roles
      data = Object.assign({}, Template[templateName].getHelpers(), {includeReplacement})

      const cheerioPotentially = parsedTemplates.find(template => template.templateName === templateName)

      let template;
      if (!cheerioPotentially) {
        console.log("Gandecki templateName", templateName);
        template = '<div></div>'
      } else {
        template = cheerioPotentially.cheerio
      }
      // console.log(`${templateName} - ${template}`);

      let myRegexp = /{{ ?#(?!.*if|.*unless|.*each|.*with)([^ }]*).*/g

      let match = myRegexp.exec(template);

      let matchedInsideTemplateNames = []

      while (match !== null) {
        // matched text: match[0]
        // match start: match.index
        // capturing group n: match[n]
        matchedInsideTemplateNames.push(match[1].trim())
        match = myRegexp.exec(template);
      }


      let matchedInsideTemplates = matchedInsideTemplateNames.map(name => {
        let cheerioInside = parsedTemplates.find(template => template.templateName === name).cheerio
        let templateInside = cheerioInside
        return {templateInside, name}
      })


        return toHTML(data, template, templateName, matchedInsideTemplates);
    }
    return beautifyHtml(includeReplacement(templateName))
}

export default renderBlazeWithTemplates;
