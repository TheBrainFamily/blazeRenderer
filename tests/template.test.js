import renderBlaze, { renderBlazeWithData, parseTemplates } from '../blazeRenderer/renderBlaze';
import returnAllTemplates from '../blazeRenderer/returnAllTemplates';

describe('template', function () {
    it(' renders properly', function () {
        const helpers = {
            hello: "is it me you looking for?",
            nope: function () {
                return "from a function"
            },
            favoriteColors: [{color: "yellow"}, {color: "blue"}, {color: "red"}]
        };
        const html = renderBlazeWithData('imports/client/lib/main.html', 'testTemplate', helpers)
        expect(html).toMatchSnapshot()
    })
})

it('parse templates', () => {
    expect(parseTemplates(returnAllTemplates('imports/'))).toMatchSnapshot()
})

it('renders templates included cross-files', () => {
    expect(renderBlaze('templateFromExternalFile')).toMatchSnapshot()

})

it('renders templates in file without data implicitly passed to it', () => {
    require('./main')
    expect(renderBlaze('testTemplate')).toMatchSnapshot()
})

it('renders template with onCreated callback and using Template.instance()', () => {
    require('./onCreatedTemplate')
    expect(renderBlaze('onCreatedTemplate')).toMatchSnapshot()
})

it('renders template with onCreated callback and using Template.instance() with two templates, one nested', () => {
  require('./nestedWithInstance')
  expect(renderBlaze('parentWithInstance')).toMatchSnapshot()
})

it('renders nested template with params and this.data in onCreated', () => {
  require('./nestedTemplateWithParameters')

  expect(renderBlaze('parentTemplate')).toMatchSnapshot()
})


it('renders template with with', () => {
  require('./withWith')

  expect(renderBlaze('withWith')).toMatchSnapshot()
})

it('renders properly each inside each inside each', () => {
  require('./eachInsideEachInsideEach')

  expect(renderBlaze('eachInsideEachInsideEach')).toMatchSnapshot()
})

it('should have access to functions added by registerHelper', () => {
  require('./registeredHelper')

  expect(renderBlaze('forRegisteredHelper')).toMatchSnapshot()
})

it('renders templateWithContentBlock', () => {
  require('./templateWithContentBlock')
  expect(renderBlaze('templateWithContentBlockOut')).toMatchSnapshot()
})

it('helpers take arguments properly', () => {
  require('./helpersWithArguments')
  expect(renderBlaze('helpersWithArguments')).toMatchSnapshot()
})

it('template pass arguments properly', () => {
	require('./passDataToTemplate')
	expect(renderBlaze('passDataToTemplate')).toMatchSnapshot()
})

it('template renders contentBlock in right order', () => {
  require('./toTable')
  expect(renderBlaze('toTable')).toMatchSnapshot()
})

it('should execute helper function inside template when calling {{helperName}} not a passed parameter that has the same name - {{helperName}}', () => {
  require('./renderMultipleSelect')
  expect(renderBlaze('testMultiple')).toMatchSnapshot()
})
//TODO need a test for skipping the each on undefined.

//TODO need a test for helper with a value of undefined

//TODO need a test for multiline {{#templateName
