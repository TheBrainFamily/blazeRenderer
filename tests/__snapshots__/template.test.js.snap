// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers take arguments properly 1`] = `"inside hello outside"`;

exports[`parse templates 1`] = `
Array [
  Object {
    "cheerio": "",
    "templateFile": "imports/client/lib/additionalTemplates.html",
    "templateName": "additionalOne1",
  },
  Object {
    "cheerio": "From one 2
    {{{ includeReplacement 'nestedTemplate'   }}}",
    "templateFile": "imports/client/lib/additionalTemplates.html",
    "templateName": "templateFromExternalFile",
  },
  Object {
    "cheerio": "",
    "templateFile": "imports/client/lib/additionalTemplates.html",
    "templateName": "additionalOne3",
  },
  Object {
    "cheerio": "Przesylac przesyla ale nie moge eachowac ponizej. Each nie dostaje zmiennej tak jabky. Tak sie dzieje tylko w table.
    Przy include dziala.
    {{#mineTable passedObjectOne=tableRowData abc=\\"def\\"}}
    {{#each tableRowData}}
    <tr>
        <td>{{name}}</td>
        <td>{{number}}</td>
        <td>{{{ includeReplacement 'someOtherTemplate'  dates.month }}}</td>
        {{#if true}}
        <td>Am I missing html tags?</td>
        {{else}}
        <td>Should not be displayed</td>
        {{/if}}
    </tr>
    {{/each}}

    {{/mineTable}}",
    "templateFile": "imports/client/lib/contentBlockTable.html",
    "templateName": "toTable",
  },
  Object {
    "cheerio": "<table>
        abc
        {{#each passedObjectOne}}
        <td class=\\"passedObject\\">
            <span class=\\"insidePsased\\">{{number}}</span>
        </td>
        {{/each}}
        <td class=\\"outsideeach\\">
            {{{ includeReplacement 'label'  }}}
        </td>
        <tbody class=\\"contentBlock\\">
        {{ Template.contentBlock }}
        </tbody>
    </table>",
    "templateFile": "imports/client/lib/contentBlockTable.html",
    "templateName": "mineTable",
  },
  Object {
    "cheerio": "<div>I am printing the MONTH: {{printThisHelper _myOwnThis}}</div>",
    "templateFile": "imports/client/lib/contentBlockTable.html",
    "templateName": "someOtherTemplate",
  },
  Object {
    "cheerio": "<span> SOME BEAUTIFUL LABEL</span>",
    "templateFile": "imports/client/lib/contentBlockTable.html",
    "templateName": "label",
  },
  Object {
    "cheerio": "<div>
        each in each
        {{#each thing}}
        <div class=\\"outsideNsted\\">{{hasLabel}}</div>
        <div class=\\"aroundNested\\">{{{ includeReplacement 'inEach'  thingInThing=thingInThing hasLabel=hasLabel }}}</div>
        {{/each}}
    </div>",
    "templateFile": "imports/client/lib/eachInEach.html",
    "templateName": "eachInEach",
  },
  Object {
    "cheerio": "inEach template
    <div class=\\"inEachFirst\\">{{hasLabel}}</div>
    {{#each inEachThing argument}}
    {{#if false}}
    if true
    {{else}}
    if false
    <div class=\\"inEachLast\\">{{hasLabel}}</div>
    {{/if}}
    {{/each}}",
    "templateFile": "imports/client/lib/eachInEach.html",
    "templateName": "inEach",
  },
  Object {
    "cheerio": "<div>
        {{#each forEachOneFirst}}
        <div><span>first {{test}}</span>
        {{#each forEachOneSecond}}
        <div><span>second {{test}}</span>
        {{#each forEachOneThird}}
        <div><span>third {{test}}</span>
        {{#each forEachOneFourth}}
            <div><span>fourth {{test}}</span></div>
        {{/each}}
        </div>
        {{/each}}
        </div>
        {{/each}}
        </div>
        {{/each}}
    </div>",
    "templateFile": "imports/client/lib/eachInsideEachInsideEach.html",
    "templateName": "eachInsideEachInsideEach",
  },
  Object {
    "cheerio": "{{say \\"hello\\"}}",
    "templateFile": "imports/client/lib/helpersWithArguments.html",
    "templateName": "helpersWithArguments",
  },
  Object {
    "cheerio": "<div>
        <div>{{ hello}} {{nope}}</div>
        {{{ includeReplacement 'nestedTemplate'   }}}
        <ul>
            {{#each favoriteColors}}
            {{{ includeReplacement 'colorRow'   }}}
            {{/each}}
        </ul>

        {{#if thisIsTrue}}
            <p>This is true (if true)</p>
            {{/if}}
            {{#unless thisIsNotTrue}}
            <p>Should show up - (unless false)</p>
            {{/unless}}
            {{#if thisIsNotTrue}}
            <p>This shouldn't show up (if false)</p>
            {{else}}
            <p>but this should show up (else)</p>
            {{/if}}
    </div>",
    "templateFile": "imports/client/lib/main.html",
    "templateName": "testTemplate",
  },
  Object {
    "cheerio": "<div>this is from nested template</div>",
    "templateFile": "imports/client/lib/main.html",
    "templateName": "nestedTemplate",
  },
  Object {
    "cheerio": "<li style=\\"color:{{color}}\\">{{ color }}</li>",
    "templateFile": "imports/client/lib/main.html",
    "templateName": "colorRow",
  },
  Object {
    "cheerio": "<select {{isMultiple}}>
        <option value=\\"testMultiple\\">testMultiple</option>
    </select>",
    "templateFile": "imports/client/lib/multipleFlagTest.html",
    "templateName": "renderMultipleSelect",
  },
  Object {
    "cheerio": "{{{ includeReplacement 'renderMultipleSelect'  isMultiple=true }}}",
    "templateFile": "imports/client/lib/multipleFlagTest.html",
    "templateName": "testMultpile",
  },
  Object {
    "cheerio": "<div>Parent: {{{ includeReplacement 'nestedWithParams'  param=hello test=\\"not hello\\" }}}</div>
    {{#each returnObject}}
    <div>Parent with object params: {{{ includeReplacement 'anotherNestedWithObjectParam'  thing.inside.object }}}</div>
    {{/each}}",
    "templateFile": "imports/client/lib/nestedTemplateWithParameters.html",
    "templateName": "parentTemplate",
  },
  Object {
    "cheerio": "<div>Child: {{param}} {{test}} Param Again - {{almostReactivelyWithParentParam}}</div>",
    "templateFile": "imports/client/lib/nestedTemplateWithParameters.html",
    "templateName": "nestedWithParams",
  },
  Object {
    "cheerio": "<div>{{_myOwnThis}}</div>",
    "templateFile": "imports/client/lib/nestedTemplateWithParameters.html",
    "templateName": "anotherNestedWithObjectParam",
  },
  Object {
    "cheerio": "<div>{{fromParentTemplateInstance}}
        {{{ includeReplacement 'nestedWithInstance'   }}}
    {{fromParentTemplateInstance}}</div>",
    "templateFile": "imports/client/lib/nestedWithInstance.html",
    "templateName": "parentWithInstance",
  },
  Object {
    "cheerio": "<div>{{fromNestedTemplateInstance}}</div>",
    "templateFile": "imports/client/lib/nestedWithInstance.html",
    "templateName": "nestedWithInstance",
  },
  Object {
    "cheerio": "<div>{{fromTemplateInstance}}</div>",
    "templateFile": "imports/client/lib/onCreatedTemplate.html",
    "templateName": "onCreatedTemplate",
  },
  Object {
    "cheerio": "{{#with dataSource}}
        <div>
            should display \\"some value\\" below
            {{{ includeReplacement 'templateToPassDataTo'  key.value }}}
        </div>
    {{/with}}",
    "templateFile": "imports/client/lib/passDataToTemplate.html",
    "templateName": "passDataToTemplate",
  },
  Object {
    "cheerio": "<span>should displayed passed argument:</span>
    <span>this directly - {{_myOwnThis}}</span>
    <span>this through helper - {{displayHelper _myOwnThis}}</span>",
    "templateFile": "imports/client/lib/passDataToTemplate.html",
    "templateName": "templateToPassDataTo",
  },
  Object {
    "cheerio": "{{#if areNotEqual 'test' 'abc'}}
    <div>
        <p>Register helper working properly!!!</p>
    </div>
    {{/if}}",
    "templateFile": "imports/client/lib/registeredHelper.html",
    "templateName": "forRegisteredHelper",
  },
  Object {
    "cheerio": "<div>External
    {{#templateWithContentBlockInside passed=innerBlockData someString=\\"someString\\"}}
        <div>div from the parent template inside the nested {{someString}}</div>
        {{#each list}}
        <p>{{something}}</p>
        {{/each}}
    {{/templateWithContentBlockInside}}
    </div>",
    "templateFile": "imports/client/lib/templateWithContentBlock.html",
    "templateName": "templateWithContentBlockOut",
  },
  Object {
    "cheerio": "<div>div from nested template {{myOwnHelper 'sdf'}}</div>
        <span>should be each below on passed argument</span>
            {{#each passed}}
            <span>{{arg}}</span>
            {{/each}}

        <span>should be content block below</span>
            <div>
            {{ Template.contentBlock }}
            </div>",
    "templateFile": "imports/client/lib/templateWithContentBlock.html",
    "templateName": "templateWithContentBlockInside",
  },
  Object {
    "cheerio": "<div>
        outside with
        {{#with myBeautifulObject}}
            with from an object {{ beautifulValue }}
            {{#with beautifulValue}}
            Using object
                this directly {{ _myOwnThis }}
                through helper {{ throughHelper _myOwnThis }}
            {{/with}}
        {{/with}}
        outside again
    </div>",
    "templateFile": "imports/client/lib/withWith.html",
    "templateName": "withWith",
  },
]
`;

exports[`renders nested template with params and this.data in onCreated 1`] = `
"<div>Parent:
    <div>Child: is it? not hello Param Again - is it?</div>
</div>

<div>Parent with object params:
    <div>thing</div>
</div>

<div>Parent with object params:
    <div>another</div>
</div>"
`;

exports[`renders properly each inside each inside each 1`] = `
"<div>

    <div><span>first abc</span>

        <div><span>second ghi</span>

            <div><span>third mno</span>

                <div><span>fourth stv</span></div>

                <div><span>fourth xyz</span></div>

            </div>

            <div><span>third pqr</span>

                <div><span>fourth stv</span></div>

                <div><span>fourth xyz</span></div>

            </div>

        </div>

        <div><span>second jkl</span>

            <div><span>third mno</span>

                <div><span>fourth stv</span></div>

                <div><span>fourth xyz</span></div>

            </div>

            <div><span>third pqr</span>

                <div><span>fourth stv</span></div>

                <div><span>fourth xyz</span></div>

            </div>

        </div>

    </div>

    <div><span>first def</span>

        <div><span>second ghi</span>

            <div><span>third mno</span>

                <div><span>fourth stv</span></div>

                <div><span>fourth xyz</span></div>

            </div>

            <div><span>third pqr</span>

                <div><span>fourth stv</span></div>

                <div><span>fourth xyz</span></div>

            </div>

        </div>

        <div><span>second jkl</span>

            <div><span>third mno</span>

                <div><span>fourth stv</span></div>

                <div><span>fourth xyz</span></div>

            </div>

            <div><span>third pqr</span>

                <div><span>fourth stv</span></div>

                <div><span>fourth xyz</span></div>

            </div>

        </div>

    </div>

</div>"
`;

exports[`renders template with onCreated callback and using Template.instance() 1`] = `"<div>value of reactive variable</div>"`;

exports[`renders template with onCreated callback and using Template.instance() with two templates, one nested 1`] = `
"<div>parent - value of reactive variable
    <div>nested - value of reactive variable</div>
    parent - value of reactive variable</div>"
`;

exports[`renders template with with 1`] = `
"<div>
    outside with with from an object really beautiful Using object this directly really beautiful through helper really beautiful outside again
</div>"
`;

exports[`renders templateWithContentBlock 1`] = `
"<div>External
    <div>div from nested template this is from my own helper sdf someString</div>
    <span>should be each below on passed argument</span>

    <span>first</span>

    <span>second</span>


    <span>should be content block below</span>
    <div>

        <div>div from the parent template inside the nested someString</div>

        <p>args</p>

        <p>in</p>

        <p>list</p>


    </div>
</div>"
`;

exports[`renders templates in file without data implicitly passed to it 1`] = `
"<div>
    <div>is it me you looking for? from a function</div>
    <div>this is from nested template</div>
    <ul>

        <li style=\\"color:yellow\\">yellow</li>

        <li style=\\"color:blue\\">blue</li>

        <li style=\\"color:red\\">red</li>

    </ul>


    <p>This is true (if true)</p>


    <p>Should show up - (unless false)</p>


    <p>but this should show up (else)</p>

</div>"
`;

exports[`renders templates included cross-files 1`] = `
"From one 2
<div>this is from nested template</div>"
`;

exports[`should have access to functions added by registerHelper 1`] = `
"<div>
    <p>Register helper working properly!!!</p>
</div>"
`;

exports[`template  renders properly 1`] = `
"<div>
    <div>is it me you looking for? from a function</div>

    <div>this is from nested template</div>

    <ul>


        <li style=\\"color:yellow\\">yellow</li>



        <li style=\\"color:blue\\">blue</li>



        <li style=\\"color:red\\">red</li>


    </ul>



    <p>Should show up - (unless false)</p>


    <p>but this should show up (else)</p>

</div>"
`;

exports[`template pass arguments properly 1`] = `
"<div>
    should display \\"some value\\" below
    <span>should displayed passed argument:</span>
    <span>this directly - some value</span>
    <span>this through helper - some value</span>
</div>"
`;

exports[`template renders contentBlock in right order 1`] = `
"Przesylac przesyla ale nie moge eachowac ponizej. Each nie dostaje zmiennej tak jabky. Tak sie dzieje tylko w table. Przy include dziala.
<table>
    abc

    <td class=\\"passedObject\\">
        <span class=\\"insidePsased\\">1</span>
    </td>

    <td class=\\"passedObject\\">
        <span class=\\"insidePsased\\">2</span>
    </td>

    <td class=\\"passedObject\\">
        <span class=\\"insidePsased\\">3</span>
    </td>

    <td class=\\"outsideeach\\">
        <span> SOME BEAUTIFUL LABEL</span>
    </td>
    <tbody class=\\"contentBlock\\">


        <tr>
            <td>first</td>
            <td>1</td>
            <td>
                <div>I am printing the MONTH: January</div>
            </td>

            <td>Am I missing html tags?</td>

        </tr>

        <tr>
            <td>second</td>
            <td>2</td>
            <td>
                <div>I am printing the MONTH: February</div>
            </td>

            <td>Am I missing html tags?</td>

        </tr>

        <tr>
            <td>third</td>
            <td>3</td>
            <td>
                <div>I am printing the MONTH: March</div>
            </td>

            <td>Am I missing html tags?</td>

        </tr>



    </tbody>
</table>"
`;
